src/test/fuzz/addrman.cpp

```diff
@@ -35,83 +35,71 @@ FUZZ_TARGET_INIT(addrman, initialize_addrman)
+            },
+            [&] {
+                addr_man.ResolveCollisions();
+            },
+            [&] {
+                (void)addr_man.SelectTriedCollision();
+            },
+            [&] {
+                (void)addr_man.Select(fuzzed_data_provider.ConsumeBool());
+            },
+            [&] {
+                (void)addr_man.GetAddr();
+            },
+            [&] {
                 const std::optional<CAddress> opt_address = ConsumeDeserializable<CAddress>(fuzzed_data_provider);
-                if (!opt_address) {
-                    break;
+                const std::optional<CNetAddr> opt_net_addr = ConsumeDeserializable<CNetAddr>(fuzzed_data_provider);
+                if (opt_address && opt_net_addr) {
+                    addr_man.Add(*opt_address, *opt_net_addr, fuzzed_data_provider.ConsumeIntegralInRange<int64_t>(0, 100000000));
                 }
-                addresses.push_back(*opt_address);
-            }
-            const std::optional<CNetAddr> opt_net_addr = ConsumeDeserializable<CNetAddr>(fuzzed_data_provider);
-            if (opt_net_addr) {
-                addr_man.Add(addresses, *opt_net_addr, fuzzed_data_provider.ConsumeIntegralInRange<int64_t>(0, 100000000));
-            }
-            break;
-        }
-        case 7: {
-            const std::optional<CService> opt_service = ConsumeDeserializable<CService>(fuzzed_data_provider);
-            if (opt_service) {
-                addr_man.Good(*opt_service, fuzzed_data_provider.ConsumeBool(), ConsumeTime(fuzzed_data_provider));
-            }
-            break;
-        }
-        case 8: {
-            const std::optional<CService> opt_service = ConsumeDeserializable<CService>(fuzzed_data_provider);
-            if (opt_service) {
-                addr_man.Attempt(*opt_service, fuzzed_data_provider.ConsumeBool(), ConsumeTime(fuzzed_data_provider));
-            }
-            break;
-        }
-        case 9: {
-            const std::optional<CService> opt_service = ConsumeDeserializable<CService>(fuzzed_data_provider);
-            if (opt_service) {
-                addr_man.Connected(*opt_service, ConsumeTime(fuzzed_data_provider));
-            }
-            break;
-        }
-        case 10: {
-            const std::optional<CService> opt_service = ConsumeDeserializable<CService>(fuzzed_data_provider);
-            if (opt_service) {
-                addr_man.SetServices(*opt_service, ServiceFlags{fuzzed_data_provider.ConsumeIntegral<uint64_t>()});
-            }
-            break;
-        }
-        case 11: {
-            (void)addr_man.Check();
-            break;
-        }
-        }
+            },
+            [&] {
+                std::vector<CAddress> addresses;
+                while (fuzzed_data_provider.ConsumeBool()) {
+                    const std::optional<CAddress> opt_address = ConsumeDeserializable<CAddress>(fuzzed_data_provider);
+                    if (!opt_address) {
+                        return;
```

Comment:
These are lambda functions, non section-end `break`s were there to allow for early termination of the test, which no longer apply as it's no longer within a `switch` statement. That functionality is now done with a `return` instead of `break`, other [examples](https://github.com/dashpay/dash/pull/5235/commits/658da948c43b4baef2d8135c5427285207e6941b#diff-88829fa89405c5b387841642b72b2f3784c46c363e34aac508ba156e35511904R36-R38) are present throughout the backport.

Reply:
I disagree. This `break` was for `while`, not for `switch`. And it was kept this way in the original PR https://github.com/bitcoin/bitcoin/pull/20828/files#diff-593914ce009ce7e5a7aedc18eeefa2be2f0da0eb2c4ae19cee43abf4a9fd5717R77 too btw.

Reply:
Didn't notice the while loop, my bad! Resolved in latest push!